### PR TITLE
[K12] Test and Test Score Objects K12 Additions

### LINKED ISSUE #177: [K12] Test and Test Score Objects K12 Additions
**Note: FOR INTERNAL USE ONLY. For support questions, please use the EDA Chatter group in [The Power of Us Hub](https://powerofus.force.com).
For feature requests, please use the Ideas tab in [The Power of Us Hub](https://powerofus.force.com).**

* **Do you want to request a *feature* or report a *bug*?**
Feature
* **What is the current behavior?**
Test and Test Score do not have a designated K12 page layout and list view
* **What is the expected behavior?**
Test and Test Score should have a designated K12 page layout and list view 
* **What is the motivation / use case for changing the behavior?**
consistency and best practice

### PR DESCRIPTION
**Pull request by kdy618**
# Critical Changes

# Changes
In version 1.8, we made the following changes. If you installed an earlier version or installed on top of an existing org, consider making them to your org.

- We introduced a K-12 All Tests list view on Test. If you choose to create a K-12 All Tests list view, we recommend this configuration:
    - Sharing Settings: All users can see this list view-
    - Filter by Owner: All tests
    - Display order: Test Name, Contact, Test Type, Test Date
      See [Create or Clone a List View](https://help.salesforce.com/articleView?id=customviews_lex.htm&type=5).
   - Create a K12 Test Compact Layout with fields in this order: Test Name, Contact, Test Type, Test Date

- We introduced a K-12 All Test Scores list view on Test Scores. If you choose to create a K-12 All Tests Scores list view, we recommend this configuration:
  - Sharing Settings: All users can see this list view
  - Filter by Owner: All test scores
  - Display order: Test Score Name, Test Type, Subject Area, Score, Percentile
    See [Create or Clone a List View](https://help.salesforce.com/articleView?id=customviews_lex.htm&type=5).
   - Add K-12 All Test Scores as a related list on Test.
     For more information, see [Display a New Custom Object After a Product Release](https://powerofus.force.com/s/article/EDA-Display-Cust-Obj-Post-Rel).
    - Create a K12 Test Compact Layout with fields in this order: Test Name, Contact, Test Type, Test Date

- We introduced new quick actions for Edit, New Note, Prinatble View, Change Owner, Clone, and Delete. If you want the New Case quick action to appear, you'll need to update the global quick action for New Case and change the record type as described in [Quick Action Considerations](https://help.salesforce.com/articleView?id=actions_considerations.htm&type=5).

# Issues Closed
Closes #177 
# New Metadata
Page Layouts:
- K12 Kit Test Layout
- K12 Kit Test Score Layout

List View:
- K12 All Tests
- K12 All Test Score

# Deleted Metadata

# Testing Notes
1. Trial Org: this should be tested only after the PR has been merged to master

2. Dev_Namespaced Shape + dev_org_namespaced flow: 
Pre-requisite: Go to Setup || User Interface || Global Actions || New Case
- Click Edit on "New Case"
- Update "RecordType" value from "Master" to "Incident" 
- Click save 

Steps for testing: 
- Create Test and Test Score records
- Test and Test Score page layout should match the screenshots on this quip doc https://salesforce.quip.com/5YeEArOU11bn#XSOACAgOHRH and https://salesforce.quip.com/5YeEArOU11bn#XSOACAVFf4P
- Test and Test Score list view should match the screenshots on this quip doc 
https://salesforce.quip.com/5YeEArOU11bn#XSOACAjlpPk and 
https://salesforce.quip.com/5YeEArOU11bn#XSOACAg0xPp

3. Dev_Namespaced Shape + qa_org flow: 
Pre-requisite: Go to Setup || User Interface || Global Actions || New Case
- Click Edit on "New Case"
- Update "RecordType" value from "Master" to "Incident" 
- Click save 

Steps for testing: 
- Create Test and Test Score records
- Test and Test Score page layout should match the screenshots on this quip doc https://salesforce.quip.com/5YeEArOU11bn#XSOACAgOHRH and https://salesforce.quip.com/5YeEArOU11bn#XSOACAVFf4P
- Test and Test Score list view should match the screenshots on this quip doc 
https://salesforce.quip.com/5YeEArOU11bn#XSOACAjlpPk and 
https://salesforce.quip.com/5YeEArOU11bn#XSOACAg0xPp

### CODE REVIEW

File: unpackaged/post/config/package.xml [hunk 1]
```diff
@@ -28,6 +28,8 @@
         <members>hed__Relationship__c.hed__HEDA_Relationship_Compact_Layout</members>
         <members>hed__Term_Grade__c.K12_Term_Grade_Compact_Layout</members>
         <members>hed__Term__c.hed__HEDA_Term_Compact_Layout</members>
+        <members>hed__Test__c.K12_Test_Compact_Layout</members>
+        <members>hed__Test_Score__c.K12_Test_Score_Compact_Layout</members>
```
Comment (skosgi1) [inline unpackaged/post/config/package.xml]: I can see the list views in the UI, but looks like it is not added here, am I missing anything?

Comment (kdy618) [reply unpackaged/post/config/package.xml]: @skosgi1 , the list view is in the test.object file.

File: unpackaged/post/config/objects/hed__Test_Score__c.object [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        <formFactor>Large</formFactor>
+        <skipRecordTypeSelect>false</skipRecordTypeSelect>
+        <type>Flexipage</type>
+    </actionOverrides>
+    <compactLayoutAssignment>K12_Test_Score_Compact_Layout</compactLayoutAssignment>
+    <compactLayouts>
+        <fullName>K12_Test_Score_Compact_Layout</fullName>
+        <fields>NAME</fields>
+        <fields>hed__Test_Type__c</fields>
+        <fields>hed__Test__c</fields>
```
[the remark this answers is not in the document]
Comment (kdy618) [reply unpackaged/post/config/objects/hed__Test_Score__c.object]: thank you! :)

File: unpackaged/post/config/package.xml (hunk 1 above)
Comment (kdy618) [reply unpackaged/post/config/package.xml]: thank you! :)

**Comment by lifewithryan** [comment]
@kdy618 looks like the build is failing compilation here, just wanted to give a heads up  in case you didn't see it.

**Comment by dgamboni** [comment]
Tested / Passes QE.

**Comment by jacebryan** [comment]
For the last bullet (for case quick action) printable is spelled wrong)

**Comment by jacebryan** [comment]
After testing in a K12 beta SO using the install_beta flow I would like to recommend adding some notes to the last bullet for New Case quick action:  In Global Actions you will see two options for New Case: 1 specific to the K12 managed package and 1 that is standard to Case. Edit the Global Action for the standard Case. Here you will update the record type to "Incident" and save. Once that is done, go to Publisher Layouts (under Global Actions), click edit, and drag the standard New Case quick action into the Global Publisher and save.

**Comment by kdy618** [comment]
> @kdy618 The changes look good, approving the PR. Can you remind me if we are going forward with this PR by documenting the steps to enable 'New Case' or are we doing a SPEC ? Also, do we want to update that we are adding new page layouts and flexi pages? Thanks

@skosgi1 , yep we are documenting it but I think we're providing a link on how to change the record type for the global quick action documentation that sfdc has. Good point on the page layout. 

@juliesnow , is it too late to add that we have new page layout for Test and Test Score?

**Comment by juliesnow** [comment]
@kdy618 + @jacebryan - will PM you re these updates.

**Comment by jacebryan** [comment]
@kdy618 - Is there a reason K12 created a custom New Case global action that has the same name as the standard New Case global action? Is there any possible issues associated with this?

**Comment by kdy618** [comment]
> @kdy618 - Is there a reason K12 created a custom New Case global action that has the same name as the standard New Case global action? Is there any possible issues associated with this?

@jacebryan , we're leveraging the standard out of the box new case global action. We did not create a new one.

**Comment by jacebryan** [comment]
@kdy618  - Unless I'm mistaken I am seeing a K12 specific managed New Case global action that uses the same api name as the standard New Case global action.

**Comment by kdy618** [comment]
> @kdy618 - Unless I'm mistaken I am seeing a K12 specific managed New Case global action that uses the same api name as the standard New Case global action.

@jacebryan , I created beta shape org + ran beta_install and I only saw 1 New case global action. Am I running the wrong flow/shape?

**Comment by jacebryan** [comment]
Can we confirm a few things here?

1. We've leveraged standard Quick Actions and made them available for use in the Salesforce Mobile and Lightning Experience Actions bar for the Test and Test Score page layouts: Edit, New Note, New Case, Printable View, Change Owner, Clone and Delete. 

- What about existing customers? Will they need to make any changes?

1. In order to use the standard New Case Quick Action on Test and Test Score page layouts you will need to update the record type on the New Case Global Action to "Incident". You will see two options in Global Actions for New Case: 1 specific to the K12 managed package and 1 that is managed. Edit the Global Action for the non-managed New Case. Here you will update the record type to "Incident" and save.

- This change will only need to be made by customers who have not already updated the New Case Global Action to "Incident".

1. The New Case Quick Action that SHOULD be showing on the Salesforce Mobile and Lightning Experience Actions bar on Test and Test Score page layouts should be the standard New Case Quick Action and not the custom K12 New Case Quick Action?

**Comment by jacebryan** [comment]
@juliesnow - here is language for use in the updated release notes. Feel free to do your magic :)

With the release of K12 1.8 we've leveraged standard Quick Actions and made them available for use in the Salesforce Mobile and Lightning Experience Actions bar for the Test and Test Score page layouts. Those Quick Actions include: Edit, New Note, New Case, Printable View, Change Owner, Clone and Delete.

Note:

- You will need to edit the Page Layouts for both Test and Test Score and remove the New Case quick action that is set by default. You will need to add the other New Case Quick Action in its place.
- In order to use the standard New Case Quick Action on Test and Test Score page layouts you will need to update the record type on the New Case Global Action to "Incident". You will see two options in Global Actions for New Case: 1 specific to the K12 managed package and 1 that is non-managed. Edit the Global Action for the non-managed New Case. Here you will update the record type to "Incident" and save. **_This change will only need to be made by customers who have not already updated the New Case Global Action to "Incident"._**

**Pull request merged**